From 1e4955adbcd9b3f5eaf2129e918ca057baed6520 Mon Sep 17 00:00:00 2001
From: Arcady Goldmints-Orlov <arcady@lunarg.com>
Date: Mon, 20 Feb 2023 20:02:36 -0500
Subject: [PATCH] Include <cstdint> header in Common.h

This change also cleans up some ifdef'd code for no longer supported
versions of MSVC.

Fixes: #3139
---
 glslang/Include/Common.h | 19 ++-----------------
 1 file changed, 2 insertions(+), 17 deletions(-)

diff --git a/glslang/Include/Common.h b/glslang/Include/Common.h
index c7f52563..080b8071 100644
--- a/glslang/Include/Common.h
+++ b/glslang/Include/Common.h
@@ -44,6 +44,7 @@
 #else
 #include <cmath>
 #endif
+#include <cstdint>
 #include <cstdio>
 #include <cstdlib>
 #include <list>
@@ -66,7 +67,7 @@ std::string to_string(const T& val) {
 }
 #endif
 
-#if (defined(_MSC_VER) && _MSC_VER < 1900 /*vs2015*/) || (defined(MINGW_HAS_SECURE_API) && MINGW_HAS_SECURE_API)
+#if defined(MINGW_HAS_SECURE_API) && MINGW_HAS_SECURE_API
     #include <basetsd.h>
     #ifndef snprintf
     #define snprintf sprintf_s
@@ -82,22 +83,6 @@ std::string to_string(const T& val) {
     #define UINT_PTR uintptr_t
 #endif
 
-#if defined(_MSC_VER) && _MSC_VER < 1800
-    #include <stdlib.h>
-    inline long long int strtoll (const char* str, char** endptr, int base)
-    {
-        return _strtoi64(str, endptr, base);
-    }
-    inline unsigned long long int strtoull (const char* str, char** endptr, int base)
-    {
-        return _strtoui64(str, endptr, base);
-    }
-    inline long long int atoll (const char* str)
-    {
-        return strtoll(str, NULL, 10);
-    }
-#endif
-
 #if defined(_MSC_VER)
 #define strdup _strdup
 #endif
-- 
2.47.2

